src/main/java/org/terracotta/statistics/derived/MinMaxAverage.java

```diff
@@ -50,15 +50,11 @@ public MinMaxAverage(Executor executor) {
   
   @Override
   public void event(long time, final long ... parameters) {
-    executor.execute(new Runnable() {
-
-      @Override
-      public void run() {
-        for (long max = maximum.get(); max < parameters[0] && !maximum.compareAndSet(max, parameters[0]); max = maximum.get());
-        for (long min = minimum.get(); min > parameters[0] && !minimum.compareAndSet(min, parameters[0]); min = minimum.get());
-        for (long sumBits = summation.get(); !summation.compareAndSet(sumBits, doubleToLongBits(longBitsToDouble(sumBits) + parameters[0])); sumBits = summation.get());
-        count.incrementAndGet();
-      }
+    executor.execute(() -> {
```

Comment:
We can eliminate all this complexity if we switch to using `LongAccumulator`, `DoubleAdder` and `LongAdder` here.

Reply:
Great! Done.